internal/infrastructure/kubernetes/proxy/resource.go

```diff
@@ -125,6 +132,10 @@ func expectedProxyContainers(infra *ir.ProxyInfra, deploymentConfig *egv1a1.Kube
 
 	var bootstrapConfigurations string
 
+	var proxyMetrics *egv1a1.ProxyMetrics
+	if infra.Config != nil {
```

Comment:
probably want to check for `nil` at each level - ` && infra.Config.Spec.Telemetry != nil && infra.Config.Spec.Telemetry.Metrics != nil `

Reply:
it's checked inside `GetRenderedBootstrapConfig`.

Reply:
looks like `Telemetry` is not a pointer https://github.com/envoyproxy/gateway/blob/23c91a6f903dc101b6bc362ac6c3432aec14b316/api/v1alpha1/envoyproxy_types.go#L45
since its optional, shouldnt it be a pointer

Reply:
updated, hope I don't miss anything.